state/store.go

```diff
@@ -25,16 +26,37 @@ const (
 
 //------------------------------------------------------------------------
 
-func calcValidatorsKey(height int64) []byte {
-	return []byte(fmt.Sprintf("validatorsKey:%v", height))
+const (
+	// prefixes are unique across all tm db's
+	prefixValidators      = byte(0x05)
+	prefixConsensusParams = byte(0x06)
+	prefixABCIResponses   = byte(0x07)
+)
+
+func encodeKey(buf []byte, height int64) []byte {
+	res, _ := orderedcode.Append(buf, height)
+	return res
+}
+
+func decodeKey(key []byte) int64 {
```

Comment:
We should have one decode function per key type, which also checks the prefix. This isn't extensible, since it assumes all keys will always have the same structure.